src/foam/core/MethodInfo.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.core;
+
+public interface MethodInfo extends Axiom {
+
+  public ClassInfo getClassInfo();
+  public MethodInfo setClassInfo(ClassInfo p);
+
+  public String getName();
+  public void setName(String name);
+
+  public Object get(Object obj);
+  public void set(Object obj, Object value);
+
+  public boolean getPermissionRequired();
+  public void authorize(foam.core.X x);
```

Comment:
I took these two methods from PropertyInfo as I thought that maybe we would like to constraint the call of the method remotely for authorized users. They are not used currently but I thought it would be useful to have them.

Reply:
Yeah we might want them in the future but for now let's keep them out of this while you're still building this out. We need to be sure not to turn anything on until we've tightened up the security around but but for the time being, we'll focus on making the ORB work.

FYI the `authorize` method comes from the Expr interface which this Axiom probably won't implement.